src/App.tsx

```diff
@@ -14,11 +20,146 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+const delicious:string[] = [
+  'Appetizing',
+  'Flavorful',
+  'Fresh',
+  'Delish',
+  'Delectable',
+  'Tasteful',
+  'Tasty',
+  'Toothsome',
+  'Yummy',
+  'Flavorful',
+];
+
+type State = {
+  selectedGoods: string[],
+};
+
+class App extends React.Component<{}, State> {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  reset = () => {
+    this.setState({ selectedGoods: [] });
+  };
+
+  add = (el:string, selectedGoods:string[]) => {
+    if (!selectedGoods.includes(el)) {
+      this.setState(prevState => (
+        { selectedGoods: [...prevState.selectedGoods, el] }));
+    }
+  };
+
+  remove = (el:string, selectedGoods:string[]) => {
+    if (selectedGoods.includes(el)) {
+      this.setState(prevState => ({
+        selectedGoods: [...prevState.selectedGoods].filter(
+          product => product !== el,
+        ),
+      }));
+    }
+  };
+
+  titleMaker = (goods:string[]) => {
+    return (
+      goods.length
+        ? (
+          <h1 className="productCards__cartContent">
+            {goods.length > 1
+              ? `${goods.slice(0, -1).join(', ')} and ${goods[goods.length - 1]} are `
+              : (`${goods.join(', ')} is `)}
+
+            selected
+
+          </h1>
+        )
+        : <h1 className="productCards__cartContent">No goods selected</h1>);
+  };
```

Comment:
```suggestion
  createTitle/formatTitle = (goods:string[]) => {
    return (
      goods.length
        ? (
          <h1 className="productCards__cartContent">
            {goods.length > 1
              ? `${goods.slice(0, -1).join(', ')} and ${goods[goods.length - 1]} are `
              : (`${goods.join(', ')} is `)}

            selected

          </h1>
        )
        : <h1 className="productCards__cartContent">No goods selected</h1>);
  };
```
по названию должно быть понятно, что это функцию, называй ее с помощью глагола